tests/core_test.py

```diff
@@ -431,17 +431,17 @@ def new_jaxpr():
     jaxpr.eqns[0].outvars[0].aval = make_shaped_array(jnp.int32(2))
     self.assertRaisesRegex(
         core.JaxprTypeError,
-        r"Variable '.' inconsistently typed as ShapedArray(.*), "
-        r"bound as ShapedArray(.*)\n\nin equation:\n\n.:i32\[\] = sin .",
+        r"Variable '.' inconsistently typed as f32\[\], "
+        r"bound as i32\[\]\n\nin equation:\n\n.:i32\[\] = sin .",
```

Comment:
These checks are more specific than they were before. The formatting is also better IMO, since it uses the short-and-readable form of types that we prefer.